SU2_CFD/src/solvers/CPassiveScalarSolver.cpp

```diff
@@ -326,85 +359,97 @@ void CPassiveScalarSolver::SetInitialCondition(CGeometry **geometry,
 
 void CPassiveScalarSolver::SetPreconditioner(CGeometry *geometry, CSolver **solver_container,  CConfig *config) {
-  
+
   unsigned short iVar;
   unsigned long iPoint, total_index;
-  
+
   su2double  BetaInc2, Density, dRhodT, dRhodC, Temperature, Delta;
-  
+
   bool variable_density = (config->GetKind_DensityModel() == INC_DENSITYMODEL::VARIABLE);
   bool implicit         = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
-  
+
   for (iPoint = 0; iPoint < nPointDomain; iPoint++) {
-    
+
     /*--- Access the primitive variables at this node. ---*/
-    
+
     Density     = solver_container[FLOW_SOL]->GetNodes()->GetDensity(iPoint);
     BetaInc2    = solver_container[FLOW_SOL]->GetNodes()->GetBetaInc2(iPoint);
     Temperature = solver_container[FLOW_SOL]->GetNodes()->GetTemperature(iPoint);
-    
+
     unsigned short nVar_Flow = solver_container[FLOW_SOL]->GetnVar();
-    
+
     su2double SolP = solver_container[FLOW_SOL]->LinSysSol[iPoint*nVar_Flow+0];
     su2double SolT = solver_container[FLOW_SOL]->LinSysSol[iPoint*nVar_Flow+nDim+1];
-    
+
     /*--- We need the derivative of the equation of state to build the
      preconditioning matrix. For now, the only option is the ideal gas
      law, but in the future, dRhodT should be in the fluid model. ---*/
-    
+
     if (variable_density) {
       dRhodT = -Density/Temperature;
     } else {
       dRhodT = 0.0;
     }
-    
-    /*--- Passive scalars have no impact on the density. ---*/
-    
-    dRhodC = 0.0;
-    
+
+    std::vector<su2double> MolecularWeight;
+    std::vector<su2double> SpecificHeatCp;
+    unsigned short n_species_mixture = nVar + 1;
+    MolecularWeight.resize(n_species_mixture);
+    SpecificHeatCp.resize(n_species_mixture);
+
+    for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+      MolecularWeight.at(iVar) = config->GetMolecular_Weight(iVar);
+      SpecificHeatCp.at(iVar) = config->GetSpecific_Heat_Cp(iVar);
+    }
+
     /*--- Modify matrix diagonal with term including volume and time step. ---*/
-    
+
     su2double Vol = geometry->nodes->GetVolume(iPoint);
     Delta = Vol / (config->GetCFLRedCoeff_Scalar()*
                    solver_container[FLOW_SOL]->GetNodes()->GetDelta_Time(iPoint));
-    
+
     /*--- Calculating the inverse of the preconditioning matrix
      that multiplies the time derivative during time integration. ---*/
-    
+
     if (implicit) {
       // nijso: do we need to wipe the entire jacobian for preconditioning?
       //for (int i_var = 0; i_var < nVar; i_var++) {
       //  for (int j_var = 0; j_var < nVar; j_var++) {
       //    Jacobian_i[i_var][j_var] = 0.0;
       //  }
       //}
-      
+
       for (iVar = 0; iVar < nVar; iVar++) {
-        
+
         total_index = iPoint*nVar+iVar;
-        
+
         su2double c = nodes->GetSolution(iPoint,iVar);
-        
+        dRhodC = -Density*((MolecularWeight[1]-MolecularWeight[0])/(MolecularWeight[0]+(MolecularWeight[1]-MolecularWeight[0])*c));
+
```

Comment:
@bigfooted @danielmayer we should also consolidate our nomenclature and be consistent throughout code and equations. Massfractions `c` or `Y` for example?